src/components/character.rs

```diff
@@ -0,0 +1,15 @@
+use amethyst::core::ecs::{Component, DenseVecStorage};
+
+pub struct Character {
+    /* Doesn't contain data yet, this is just a marker */
+}
+
+impl Character {
+    pub fn new() -> Character {
+        Character {}
```

Comment:
my chance to just ask dumb rust questions. Is this allocated on the heap? Or is that just a dumb question and the compiler will decide based on however it's used

Reply:
The Character struct would be placed on the stack if it contained any data.
In this case it's a zero-sized struct so it would be compiled away and not
use any memory.

To put it on the Heap you would need to put it in a Box (ex:`Box::new(Character{})` )

Since we are adding this as a marker trait to entities, technically the
game engine will need to store if the entity contains it or not, but the
actual struct itself currently uses no memory.